File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/util/text/IChatComponent.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.util.text;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.utils.ReflectUtil;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.IWrapper;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.util.ChatStyle;

import java.lang.reflect.Method;

@WrapperClass(mcpName = "net.minecraft.util.IChatComponent", targetMap = MapEnum.VANILLA189)
public class IChatComponent extends IWrapper {
    @WrapMethod(mcpName = "getUnformattedText", targetMap = MapEnum.VANILLA189)
    public static Method getUnformattedText;
    @WrapMethod(mcpName = "getFormattedText", targetMap = MapEnum.VANILLA189)
    public static Method getFormattedText;
    @WrapMethod(mcpName = "getChatStyle", targetMap = MapEnum.VANILLA189)
    public static Method getChatStyle;

    public IChatComponent(Object obj) {
        super(obj);
    }

    public String getUnformattedText() {
        return (String) ReflectUtil.invoke(getUnformattedText, getWrapObject());
    }

    public String getFormattedText() {
        return (String) ReflectUtil.invoke(getFormattedText, getWrapObject());
    }

    public ChatStyle getChatStyle() {
        return new ChatStyle(invoke(getChatStyle));
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/util/text/IChatComponent.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.util.text

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.annotation.WrapMethod
import rbq.wtf.lycoris.client.wrapper.annotation.WrapperClass
import rbq.wtf.lycoris.client.wrapper.IWrapper
import rbq.wtf.lycoris.client.wrapper.wrappers.util.ChatStyle
import java.lang.reflect.Method

@WrapperClass(mcpName = "net.minecraft.util.IChatComponent", targetMap = MapEnum.VANILLA189)
class IChatComponent(obj: Any) : IWrapper(obj) {
    companion object {
        @WrapMethod(mcpName = "getUnformattedText", targetMap = MapEnum.VANILLA189)
        lateinit var getUnformattedText: Method

        @WrapMethod(mcpName = "getFormattedText", targetMap = MapEnum.VANILLA189)
        lateinit var getFormattedText: Method

        @WrapMethod(mcpName = "getChatStyle", targetMap = MapEnum.VANILLA189)
        lateinit var getChatStyle: Method
    }

    val unformattedText: String
        get() = invoke(getUnformattedText) as String
    val formattedText: String
        get() = invoke(getFormattedText) as String
    val chatStyle: ChatStyle
        get() = ChatStyle(invoke(getChatStyle)!!)
}